# Ex3 - Merge

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [1]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [16]:
#1.
data1 = pd.Series(np.random.randint(1,5, size=100), name="1to4")

data2 = pd.Series(np.random.randint(1,4, size=100), name="1to3")

data3 = pd.Series(np.random.randint(10000, 30001, size=100), name="10kto30k")

0     13348
1     12940
2     14378
3     22381
4     21622
      ...  
95    26510
96    25530
97    23760
98    28933
99    26274
Name: 10kto30k, Length: 100, dtype: int64

### Step 3. Let's create a DataFrame by joinning the Series by column

In [19]:
totals = pd.concat([data1, data2, data3], axis=1)

totals.head()

,1to4,1to3,10kto30k
0,1,2,13348
1,3,2,12940
2,4,2,14378
3,2,2,22381
4,2,3,21622


### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [33]:
house_info = totals.rename(columns={'1to4':'bedrs', '1to3':'bathrs', '10kto30k':'price_sqr_meter'})

house_info.head()

,bedrs,bathrs,price_sqr_meter
0,1,2,13348
1,3,2,12940
2,4,2,14378
3,2,2,22381
4,2,3,21622


### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [43]:
bigcolumn = pd.concat([data1, data2, data3], axis=0, ignore_index=True).to_frame()

bigcolumn.size

300

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [40]:
#Al indicar ignore_index en el ejercicio 5 este problema se soluciona.
bigcolumn.tail(1)

299    26274
dtype: int64

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [41]:
#Al indicar ignore_index en el ejercicio 5 este problema se soluciona.
bigcolumn.tail(1)

299    26274
dtype: int64